space.go

```diff
@@ -168,6 +168,20 @@ func (p Positionf) Move(to Positionf, speed Positionf) Positionf {
 	return p
 }
 
+// CrossProduct calculates the 2D cross product (determinant) of vectors p->p1 and p1->p2 to determine their orientation.
+func (p *Positionf) CrossProduct(p1, p2 *Positionf) float32 {
+	return (p1.X-p.X)*(p2.Y-p1.Y) - (p1.Y-p.Y)*(p2.X-p1.X)
+}
```

Comment:
What's the point of having `p` here? The cross product should be between `p` and other position/vector:

```go
func (p *Positionf) CrossProduct(other *Positionf) float32
```

Reply:
Ups, forget my comment. I see you are using it and vectors are relative to the points.